app/uk/gov/hmrc/customs/declaration/services/DeclarationStatusService.scala

```diff
@@ -66,9 +69,9 @@ class DeclarationStatusService @Inject()(override val logger: DeclarationsLogger
                 Left(ErrorGenericBadRequest.XmlResult.withConversationId)
             }
           }).recover{
-          case e: HttpException if e.responseCode == 404 =>
-            logger.warn(s"declaration status call failed with 404: ${e.getMessage}")
-            Left(ErrorResponse.ErrorNotFound.XmlResult.withConversationId)
+          case e: HttpException if isNotFoundOrForbidden(e.responseCode) =>
```

Comment:
@kevin-co-hmrc Because intellji said complexity exceed 11, max 10.

Reply:
Understood. However, I think we can reduce cyclomatic complexity _and_ make the code easier to read. One possible way is to extract out the PartialFunction in `.recover`, and then have one `case` clause for each status code there. What do you think?

Reply:
I am not sure, as other than 403 or 404 and other kind of throwable exceptions wants to fall into NonFatal case. So this spilts into 2 groups. Can you please explain to me why you think it's hard to read?

Reply:
Sorry, I'm not being very clear.

To illustrate, my suggestion is, instead of:

```scala
  .recover{
          case e: HttpException if isNotFoundOrForbidden(e.responseCode) =>
            logger.warn(s"declaration status call failed with ${e.responseCode}: ${e.getMessage}")
            Left(deriveErrorResponse(e.responseCode).withConversationId)
          case NonFatal(e) =>
            logger.error(s"declaration status call failed: ${e.getMessage}", e)
            Left(ErrorResponse.ErrorInternalServerError.XmlResult.withConversationId)
        }...

  private val isNotFoundOrForbidden = (statusCode: Int) => statusCode == 404 ||
    (configService.declarationsConfig.payloadForbiddenEnabled && statusCode == 403)

  private def deriveErrorResponse(statusCode: Int): Result = {
    statusCode match {
      case FORBIDDEN =>
        ErrorResponse.ErrorPayloadForbidden.XmlResult
      case NOT_FOUND =>
        ErrorResponse.ErrorNotFound.XmlResult
    }
  }
```

we do something like:

```scala
  .recover(recoverErrors)...

  private def recoverErrors[A](implicit ar: AuthorisedRequest[A]): PartialFunction[Throwable, Left[Result, Nothing]] = {
    case e: HttpException if e.responseCode == 404 =>
      logger.warn(s"declaration status call failed with 404: ${e.getMessage}")
      Left(ErrorResponse.ErrorNotFound.XmlResult.withConversationId)
    case e: HttpException if e.responseCode == 403 && configService.declarationsConfig.payloadForbiddenEnabled =>
      logger.warn(s"declaration status call failed with 403: ${e.getMessage}")
      Left(ErrorResponse.ErrorPayloadForbidden.XmlResult.withConversationId)
    case NonFatal(e) =>
      logger.error(s"declaration status call failed: ${e.getMessage}", e)
      Left(ErrorResponse.ErrorInternalServerError.XmlResult.withConversationId)
  }
```

Reply:
Think about it, that is like doing 2 conditional check, no? Please not really like if e.responseCode == FORBIDDEN && configService.declarationsConfig.payloadForbiddenEnabled in case if condition. I am taking some stuff out from the send method as it's over 25 lines.